PowerEditor/src/WinControls/Preference/preferenceDlg.cpp

```diff
@@ -3548,6 +3474,295 @@ intptr_t CALLBACK LanguageSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPARA
+					::EnableWindow(::GetDlgItem(_hSelf, IDC_EDIT_TABSIZEVAL), !useDefaultTab);
+					::EnableWindow(::GetDlgItem(_hSelf, IDC_RADIO_REPLACEBYSPACE), !useDefaultTab);
+					::EnableWindow(::GetDlgItem(_hSelf, IDC_RADIO_USINGTAB), !useDefaultTab);
+					::EnableWindow(::GetDlgItem(_hSelf, IDC_CHECK_BACKSPACEUNINDENT), !useDefaultTab);
+
+					// write in langs.xml
+					if (useDefaultTab)
+						nppParam.insertTabInfo(lang->getLangName(), -1, false);
+
+					redrawDlgItem(IDC_TABSIZE_STATIC);
+					redrawDlgItem(IDC_INDENTUSING_STATIC);
+
+					return TRUE;
+				}
+
+				case IDC_RADIO_AUTOINDENT_NONE:
+				{
+					nppGUI._maintainIndent = autoIndent_none;
+					return TRUE;
+				}
+				case IDC_RADIO_AUTOINDENT_BASIC:
+				{
+					nppGUI._maintainIndent = autoIndent_basic;
+					return TRUE;
+				}
+				case IDC_RADIO_AUTOINDENT_ADVANCE:
+				{
+					nppGUI._maintainIndent = autoIndent_advance;
+					return TRUE;
+				}
+
+				default:
+				{
+					break;
+				}
+			}
+		}
+		[[fallthrough]];
+		default:
+		{
+			break;
+		}
+	}
+	return FALSE;
+}
+
+
+intptr_t CALLBACK LanguageSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM lParam)
+{
+	NppParameters& nppParam = NppParameters::getInstance();
+	NppGUI & nppGUI = nppParam.getNppGUI();
+	NativeLangSpeaker *pNativeSpeaker = nppParam.getNativeLangSpeaker();
+
+	switch (message) 
+	{
+		case WM_INITDIALOG :
+		{
+			//
+			// Lang Menu
+			//
+			for (int i = L_TEXT ; i < nppParam.L_END ; ++i)
+			{
+				wstring str;
+				if (static_cast<LangType>(i) != L_USER)
+				{
+					int cmdID = nppParam.langTypeToCommandID(static_cast<LangType>(i));
+					if ((cmdID != -1))
+					{
+						getNameStrFromCmd(cmdID, str);
+						if (str.length() > 0)
+						{
+							_langList.push_back(LangMenuItem(static_cast<LangType>(i), cmdID, str));
+						}
+					}
+				}
+			}
+
+			std::sort(_langList.begin(), _langList.end());
+
+			for (size_t i = 0, len = _langList.size(); i < len; ++i)
+			{
+				::SendDlgItemMessage(_hSelf, IDC_LIST_ENABLEDLANG, LB_ADDSTRING, 0, reinterpret_cast<LPARAM>(_langList[i]._langName.c_str()));
+			}
+
+			std::sort(nppGUI._excludedLangList.begin(), nppGUI._excludedLangList.end());
+				
+			for (size_t i = 0, len = nppGUI._excludedLangList.size(); i < len ; ++i)
+			{
+				::SendDlgItemMessage(_hSelf, IDC_LIST_DISABLEDLANG, LB_ADDSTRING, 0, reinterpret_cast<LPARAM>(nppGUI._excludedLangList[i]._langName.c_str()));
+			}
+
+			::SendDlgItemMessage(_hSelf, IDC_CHECK_LANGMENUCOMPACT, BM_SETCHECK, nppGUI._isLangMenuCompact?BST_CHECKED:BST_UNCHECKED, 0);
+			::EnableWindow(::GetDlgItem(_hSelf, IDC_BUTTON_REMOVE), FALSE);
+			::EnableWindow(::GetDlgItem(_hSelf, IDC_BUTTON_RESTORE), FALSE);
+
+			return TRUE;
+		}
+
+		case WM_CTLCOLOREDIT:
+		{
+			return NppDarkMode::onCtlColorSofter(reinterpret_cast<HDC>(wParam));
+		}
+
+		case WM_CTLCOLORLISTBOX:
+		{
+			return NppDarkMode::onCtlColorListbox(wParam, lParam);
+		}
+
+		case WM_CTLCOLORDLG:

```

Comment:
@donho 
could you add `case WM_CTLCOLORSTATIC:` below to make static texts respect dark mode.